index.js

```diff
@@ -47,14 +61,22 @@ function scan (obj, options) {
     next = []
 
     for (const node of nodes) {
-      if (Object.prototype.hasOwnProperty.call(node, '__proto__')) { // Avoid calling node.hasOwnProperty directly
-        if (options.protoAction !== 'remove') {
+      if (protoAction !== 'ignore' && Object.prototype.hasOwnProperty.call(node, '__proto__')) { // Avoid calling node.hasOwnProperty directly
+        if (protoAction === 'error') {
           throw new SyntaxError('Object contains forbidden prototype property')
         }
 
         delete node.__proto__ // eslint-disable-line
```

Comment:
Nit: I would change this eslint comment to `// eslint-disable-line no-proto` to only disable that rule and not all other rules.

(I know this isn't part of this PR, but you're changing so many other things, so why not 🤷‍♂)